File: server/ExpenseTracker/ExpenseTracker.Application/Dtos/Account/AccountRequestDto.cs
using ExpenseTracker.Application.Dtos.Money;
using ExpenseTracker.Domain.Entities;

namespace ExpenseTracker.Application.Dtos.Account;
public class AccountRequestDto
{
    public string Name { get; set; } = default!;
    public AccountType Type { get; set; }
    public MoneyRequestDto Balance { get; set; } = default!;

}

File: clients/web/expense-tracker/src/domain/models/account/AccountRequestDto.ts
import MoneyRequestDto from "../money/MoneyRequestDto";
import AccountType from "./AccountType";

export default class AccountResponseDto {
  OwnerId!: string;

  Name!: string;

  Type!: AccountType;

  Balance!: MoneyRequestDto;

  public constructor(init?: Partial<AccountResponseDto>) {
    Object.assign(this, init);
  }
}
